src/VestedRewardsDistribution.sol

```diff
@@ -181,45 +181,51 @@ contract VestedRewardsDistribution {
      */
     function distribute() external returns (uint256 amount) {
         require(vestId != INVALID_VEST_ID, "VestedRewardsDistribution/invalid-vest-id");
-        require(block.timestamp > lastDistributedAt, "VestedRewardsDistribution/no-pending-distribution");
 
-        uint256 unpaid = dssVest.unpaid(vestId);
-        uint256 maxAmount = _getMaxAmount();
-
-        dssVest.vest(vestId, maxAmount);
+        amount = _getAmount();
+        require(amount > 0, "VestedRewardsDistribution/no-pending-amount");
+        dssVest.vest(vestId, amount);
         // We are ignoring the checks-effects-interactions pattern because `dssVest` is trusted.
         lastDistributedAt = block.timestamp;
 
-        // `dssVest.vest()` sadly does not return the actual amount of vested tokens.
-        // Also it is not safe to query the gem balance of this contract because it might have dangling tokens.
-        // The easiest way is to replicate the internal logic of DssVest here.
-        amount = _min(unpaid, maxAmount);
-        require(amount > 0, "VestedRewardsDistribution/no-pending-amount");
         require(gem.transfer(address(stakingRewards), amount), "VestedRewardsDistribution/transfer-failed");
         stakingRewards.notifyRewardAmount(amount);
 
         emit Distribute(amount);
     }
 
     /**
-     * @notice Gets the max amount to pull from the vesting stream for the next distribution.
+     * @notice Gets the amount to pull from the vesting stream for the current distribution.
      * @dev If `calc` is set, it delegates the calculation to that contract.
-     *      Otherwise, it returns the default amount `type(uint256.max)`.
-     * @return The max amount of tokens.
+     *      Otherwise, it returns the unpaid vested amount.
+     * @return The amount of tokens to pull.
      */
-    function _getMaxAmount() internal view returns (uint256) {
-        if (calc == address(0)) {
-            return type(uint256).max;
+    function _getAmount() internal view returns (uint256) {
+        // We ensured that `clf == bgn`, so we could use either one.
+        uint256 clf = dssVest.clf(vestId);
+        // If the current timestamp is before the cliff, there is nothing to be distriubted.
+        if (block.timestamp < clf) {
+            return 0;
+        }
```

Comment:
This check is redundant with the unpaid check below. You can just remove it.

Reply:
Yeah, good point.